src/nginx.c

```diff
@@ -211,7 +215,11 @@ static void submit (const char *type, const char *inst, long long value)
 
   vl.values = values;
   vl.values_len = 1;
-  sstrncpy (vl.host, hostname_g, sizeof (vl.host));
+  if (hostname != NULL) {
```

Comment:
Changed. I changed the variable to host as well to keep it consistent with the config option.